README.md

```diff
@@ -162,11 +167,81 @@ If we already have a WMS GetMap URL, we can use it directly:
+
+To get the OAuth Client's id and secret, a new OAuth Client must be created in [**User settings**](https://apps.sentinel-hub.com/dashboard/#/account/settings) on **Sentinel Hub Dashboard** under **OAuth clients**.
+OAuth Client's secret is shown only before the creation process is finished so be mindful to save it.
+
+In javascript requesting the authentication token can be done with builtin XMLHttpRequest:
+
+```javascript
+const clientId = /* OAuth Client's id, best to put it in .env file and use it from there */;
+const clientSecret = /* OAuth client's secret, best to put it in .env file and use it from there */;
+
+let xhr = new XMLHttpRequest();
+xhr.open("POST", 'https://services.sentinel-hub.com/oauth/token', true);
+xhr.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
+xhr.onreadystatechange = function() {
+  if (this.readyState === XMLHttpRequest.DONE && this.status === 200) {
+    console.log(xhr.responseText);
+  }
+}
+xhr.send("grant_type=client_credentials&client_id="+clientId+"&client_secret="+clientSecret);
+```
+
+Or by making the request with Axios:
+
+```javascript
+const axios = require('axios');
+// import axios from 'axios';
+
+const clientId = /* OAuth Client's id, best to put it in .env file and use it from there */;
+const clientSecret = /* OAuth client's secret, best to put it in .env file and use it from there */;
+
+await axios({
+  method: 'post',
+  url: 'https://services.sentinel-hub.com/oauth/token',
+  headers: { 'content-type': 'application/x-www-form-urlencoded' },
+  data: "grant_type=client_credentials&client_id="+clientId+"&client_secret="+clientSecret,
+})
+  .then(function(response){
+    console.log(response.data.access_token);
+  })
+  .catch(function(error) {
+    console.log(error);
+  });
+```
+
+More about authentication is avilable at [Sentinel Hub documentation](https://docs.sentinel-hub.com/api/latest/#/API/authentication).
+
+# Examples
+This project contains some examples to demonstrate how the library is used.
+Some preparation is needed before running the examples.
+
+## Preparation before running examples
+To run the examples, the environment variables must be set.
+These variables should be put in the `.env` file in the root folder of this project.
+
+- `CLIENT_ID`: OAuth Client's id (optional, authentication is needed for examples that use Processing API)
+- `CLIENT_SECRET`: OAuth Client's secret (optional, authentication is needed for examples that use Processing API)
+
+- `INSTANCE_ID`: id of the configuration instance that will be used in examples
+- `S2L2A_LAYER_ID`: id of the Sentinel-2 L2A layer from that instance
+- `S1GRD_LAYER_ID`: id of the Sentinel-1 GRD layer from that instance
+
+Instance can be created with the [**Configurator**](https://apps.sentinel-hub.com/dashboard/#/configurations) on the **Sentinel Hub Dashboard**.
+It should contain one Sentinel-2 L2A layer and one Sentinel-1 GRD layer .
```

Comment:
Typo: space before the dot.

I would suggest rephrasing this sentence:
```
It should contain at least one Sentinel-2 L2A layer and one Sentinel-1 GRD layer, whose layer IDs should be the same as set in `S2L2A_LAYER_ID` and `S1GRD_LAYER_ID` env vars respectively.
```

Reply:
Thanks. Used your suggestion as it seems better.